trees/trees.go

```diff
@@ -13,6 +13,15 @@ import "github.com/emirpasic/gods/containers"
 
 // Tree interface that all trees implement
 type Tree interface {
+	New() Tree
+	Iterator() containers.ReverseIteratorWithKey
+	Put(key interface{}, value interface{})
+	Remove(key interface{})
+	Get(key interface{}) (interface{}, bool)
+	Keys() []interface{}
+	Min() (interface{}, interface{})
+	Max() (interface{}, interface{})
+
```

Comment:
I agree. Then redblacktree and avltree will satisfy the interface say OrderedTree and keep Tree the same as container? And then treeset/treemap etc. will have an OrderedTree as its backing store?